gsm.app v2.6.0 Release Candidate

jonthegeek opened the pull request:
# gsm.app v2.6.0

This release provides usability upgrades for the Domain Details tab. It also adds new contributor guidelines and standardized issue templates.

## Business Requirement 519: Users can easily explore Domain Details

- Users can view counts of records for each available data domain within the tab itself, rather than in a table devoted to counts. This freed up additional space for new features (#543).
- Users can filter domain data using text boxes at the top of each column of each table (#385).
- Users can visualize domain categorical variable counts by value in an interactive plot (#496).
- More rows are shown by default in the Domain Details tables (#555).

## Other changes

- The Business Requirements articles now account for "superseded" features, and allows for "deep linking" to specific features within a business requirement (#559).
- The contributing guidelines and issue templates have been standardized across the gsm projects (#560).
- The default name for apps is now smarter about which options it chooses (#432).
- The sample data (used in demo apps) has been updated to include Site Risk Score data, and the app has been updated to make use of such data.

jharmon-gilead committed:
Merge pull request #552 from Gilead-BioStats/main

main to dev

jharmon-gilead committed:
Fix 543: Move domain counts to tabs (#550)

* Move domain counts to tabs.

- Closes #543.

* Make demo apps more positron-friendly.

* Finish implementation of 543.

* Rename back to mod_DomainLabel

* Implement simple filters.

Closes #385.

* Re-document.

* Simplify mod_DomainLabel to use textOutput, not uiOutput, per Copilot review (good bot!).

I also got rid of some leftover `shiny::` namespaces.

jharmon-gilead committed:
Fix 496 Prevalence Plots (#554)

* Move domain counts to tabs.

- Closes #543.

* Make demo apps more positron-friendly.

* Finish implementation of 543.

* Rename back to mod_DomainLabel

* Implement simple filters.

Closes #385.

* Re-document.

* Simplify mod_DomainLabel to use textOutput, not uiOutput, per Copilot review (good bot!).

I also got rid of some leftover `shiny::` namespaces.

* Implement prevalence plot

Add a Prevalence panel on the Domain Details tab.

Also refactor some duplicate code to simplify and solidify.

Closes #496.

* Add packages for plots.

* Update snapshots.

The sample error never occurs, because the Prevalence Plot alwyas forces data to load at the study level. Will update this and likely change this comment.

As part of the fix, closes #504.

* Make the prevalence plot behavior feel more natural

* Abstract things a bit more neatly.

* Clean and document new functions.

* Start BR tests for (this part of) 519.

Stopping here to push so I can make sure nothing else is weird.

* Fix R CMD check.

* Fix coverage.

* Skip visual test on most machines.

* Don't use base pipe yet.

* Skip other screenshot test on CI.

* Set up 519 validation article.

* Continue 519 validation tests.

* Oops, I never updated the version to dev.

* Update BR index.

* Show 100 rows by default.

* Address review comments.

* Fix selector button value.

jharmon-gilead committed:
Implement superseded label (#559)

* Implement superseded label

When an issue is labeled "superseded," no longer display its "proof."

* Add the ability to deep-link to a specific feature, eg by adding `#523` to the URL.

* Auto-generate BR articles in setup.

lauramaxwell committed:
Update .github folder templates using github API (#560)

* Remove old issue template: bugfix.md

* Remove old issue template: feature.md

* Remove old issue template: qc.md

* Update 1-requirement.md issue template

* Update 2-bug.md issue template

* Update 3-feature.md issue template

* Update 4-technical.md issue template

* Update 5-documentation.md issue template

* Update config.yml issue template

* Delete CONTRIBUTING.md

* Update CONTRIBUTING.md

* Update workflows/r-releaser.yaml

jharmon-gilead committed:
Protect against NA names (#561)

* Protect against NA names

Also protect screenshots against a very slight change in ggplot2 4.0.0. Screenshot changes that still had to happen are imperceptible to me.

Since this does not impact any acceptance requirements of actual business requirements, it is being unit-tested, but not specifically included in the qualification suite.

- Closes #432.

* Remove protocol_title

* Tests will fail until all the other PRs (.core, .mapping, .reporting, .kri) are merged to main, but they should pass once that happens.

* Use `@dev` for gsm packages so we can merge this and prep the release.

* Use main versions.

jonthegeek committed:
gsm.app v2.6.0 Release Candidate

# gsm.app v2.6.0

This release provides usability upgrades for the Domain Details tab. It also adds new contributor guidelines and standardized issue templates.

jonthegeek commented:
https://gilead-biostats.github.io/gsm.app/dev/articles/br518.html#535 shows how #559 works (and you can navigate around that version of the site to see the business requirements for #519, too).

samussiah commented:
The latest `gsm.kri` release contains [debug code with mock data from `gsm.viz`](https://github.com/Gilead-BioStats/gsm.viz/blob/dev/src/groupOverview/deriveGroupMetrics.js#L74-L86) that needs to be removed. If we re-release `gsm.kri` after addressing this issue `gsm.app` should pull in the latest, so I think this one's good to go.

<img width="1286" height="699" alt="image" src="https://github.com/user-attachments/assets/510ba834-0f60-4a6e-a1e0-22d60221f244" />

jonthegeek commented:
> The latest `gsm.kri` release contains [debug code with mock data from `gsm.viz`](https://github.com/Gilead-BioStats/gsm.viz/blob/dev/src/groupOverview/deriveGroupMetrics.js#L74-L86) that needs to be removed. If we re-release `gsm.kri` after addressing this issue `gsm.app` should pull in the latest, so I think this one's good to go.
> 
> <img width="1286" height="699" alt="image" src="https://github.com/user-attachments/assets/510ba834-0f60-4a6e-a1e0-22d60221f244" />
> 

Yup! Some tests might break, but this widget will update, and I can do a point release to fix the tests.

jonthegeek commented:
@samussiah I'll probably want to update gsm.app demo data to include whatever is needed to include risk score. And now that I say that, I'm not sure what will happen when that's present. So I can't say for certain that it will work. What changes in the data for this? I guess I assumed it was calculated from the existing data.

samussiah commented:
> @samussiah I'll probably want to update gsm.app demo data to include whatever is needed to include risk score. And now that I say that, I'm not sure what will happen when that's present. So I can't say for certain that it will work. What changes in the data for this? I guess I assumed it was calculated from the existing data.

The `Results` data will contain additional rows representing the site risk score metric. The `GroupOverview` widget knows what to do with them, and if they're not there that site risk score column shouldn't appear. Will circle back to `gsm.viz` later today or tomorrow.

Theoretically the `sample-data` script in `gsm.app` should pull in site risk score automatically, but I'm hitting this error:

```
Error in if ((RedResults$n[RedResults$GroupLevel == "Site"] >= 5) || intSeed >  : 
  missing value where TRUE/FALSE needed
```

samussiah commented:
The CM team noticed an issue with the fetch function, captured in [this comment](https://github.com/Gilead-BioStats/gsm.app/issues/449#issuecomment-3312862490) on #449.  We should probably address this prior to release if it's not a big update.

samussiah reviewed (changes requested):
Everything's looking good on my end!  The CM team did notice [an issue](https://github.com/Gilead-BioStats/gsm.app/issues/449#issuecomment-3312862490) that we probably want to address if it's not going to cause unexpected behavior.

jonthegeek commented:
@samussiah That's a bigger problem than it looks like. I don't re-call the fetch function if we already have the associated group cached. I didn't know that we'd want to load inconsistent data. Even if we fix your fetch function, this would fail if you loaded the group first, because I'd try to subset the group.

Is such a situation ok? It seems like a data issue. I suspect a lot of other things would also break when those aren't consistent.

samussiah commented:
> @samussiah That's a bigger problem than it looks like. I don't re-call the fetch function if we already have the associated group cached. I didn't know that we'd want to load inconsistent data. Even if we fix your fetch function, this would fail if you loaded the group first, because I'd try to subset the group.
> 
> Is such a situation ok? It seems like a data issue. I suspect a lot of other things would also break when those aren't consistent.

I can see the world where we want to maintain the hierarchy at all times because some modules rely on it, like the prevalence charts. But there are also scenarios where the hierarchy doesn't matter, like in a participant profile or when invoking the fetch function. Looks like the caching strategy would require a lot of rework so it's not worth fixing ahead of this release.

samussiah reviewed (approved):
Nice work @jonthegeek!

jonthegeek commented:
> Theoretically the `sample-data` script in `gsm.app` should pull in site risk score automatically, but I'm hitting this error:
> 
> ```
> Error in if ((RedResults$n[RedResults$GroupLevel == "Site"] >= 5) || intSeed >  : 
>   missing value where TRUE/FALSE needed
> ```

The seed that I chose now results in 0 results with red flags, let alone >= 5 sites with red flags. I added a protection for this and am re-running to try to find a set of sample data that's usable for demos. gsm.datasim tends toward 0 flags so... this might take a while 🙃

jonthegeek committed:
Properly use site risk score.

This required new data, so lots of tests changed. I tried to make them more general so there will be fewer changes the next time we update the data.
